src/java.base/share/classes/java/nio/file/Path.java

```diff
@@ -248,6 +249,62 @@ public static Path of(URI uri) {
      */
     Path getFileName();
 
+    /**
+     * Returns the file name extension of this path as a {@code String}. The
+     * extension is the portion of the {@linkplain #getFileName file name}
+     * string after the last period character ('.', U+002E FULL STOP). Leading
+     * period characters in the string are ignored. If there is no  extension,
+     * then {@code null} is returned. This will occur if the path has zero
+     * elements ({@link #getFileName()} returns {@code null}), the file name
+     * string does not contain a period character, or only the first character
+     * is a period character. If the last character is a period character but
+     * some other character is not, then the extension is the
+     * {@linkplain String#isEmpty() empty} string.
+     *
+     * @implSpec
+     * The default implementation is equivalent for this path to:
+     * <pre>{@code
+     * int lastPeriod = fileName.lastIndexOf('.');
+     * if (lastPeriod <= 0)
+     *     return null;
+     * return lastPeriod == fileName.length() - 1 ?
+     *     "" :
+     *     fileName.substring(lastPeriod + 1);
```

Comment:
Can L270-272 in the implSpec be cleaned up? It's okay to use the ternary operator but would be easier to read if there is parenthesis around the condition and the ? and : weren't at the end of the lines.

Reply:
I would suggest that the code in the implSpec be removed. It doesn't add anything and it potentially confuses things. There needs to be an implSpec though since this is a default method. The essential thing to have in implSpec is self-calls on `this`. The only self-call is to `getFileName`, so I'd include in the implSpec some prose to say that `getFileName` is called and (if non-null) is converted to a String, upon which further processing is done.